src/fileio.cpp

```diff
@@ -295,20 +294,36 @@ static const char * const _subdirs[] = {
 };
 assert_compile(lengthof(_subdirs) == NUM_SUBDIRS);
 
-const char *_searchpaths[NUM_SEARCHPATHS];
-TarList _tar_list[NUM_SUBDIRS];
+/**
+ * The search paths OpenTTD could search through.
+ * At least one of the slots has to be filled with a path.
+ * An empty string tells that there is no such path for the
+ * current operating system.
+ */
+std::array<std::string, NUM_SEARCHPATHS> _searchpaths;
+std::array<TarList, NUM_SUBDIRS> _tar_list;
 TarFileList _tar_filelist[NUM_SUBDIRS];
 
 typedef std::map<std::string, std::string> TarLinkList;
 static TarLinkList _tar_linklist[NUM_SUBDIRS]; ///< List of directory links
 
+/**
+ * Checks whether the given search path is a valid search path
+ * @param sp the search path to check
+ * @return true if the search path is valid
+ */
+bool IsValidSearchPath(Searchpath sp)
+{
+	return sp < _searchpaths.size() && !_searchpaths[sp].empty();
```

Comment:
new warning, possibly Searchpath should be forced to be unsigned type?

Reply:
Ah, didn't see that as it is apparently x86 only.